### DISCUSSION TITLE
How do I use the PCM1808 with this library? and is there a way to switch inputs? (e.g. switching from A2DP to the adc to act as an wired audio input)

### DISCUSSION THREAD

**Discussion by AyceBenz** [pschatzmann/arduino-audio-tools#1796] [Q&A] [answered]
How do I use the PCM1808 with this library? and is there a way to switch inputs? (e.g. switching from A2DP to the adc to act as an wired audio input)

**Comment by pschatzmann** [comment, marked as answer]
As you can see in my Wiki, I have [never used the PCM1818](https://github.com/pschatzmann/arduino-audio-tools/wiki/External-ADC). So read it's datasheet so that you know how to use it!
Here are a couple of key points:
- most likey it only supports 24 or 32 bits
- you need to provide it a master clock (via GPIO0)
- try to find out if it can be used in slave mode (with the ESP32 as master) which is more flexible. If not you need to set the ESP32 to slave mode and to the provided sample rate from the module
- try to find out how to set the format: I suggest to prefer standard I2S which is used by default in my API, otherwise you need to define it when you configure I2S.

If you have one AudioSink and multiple sources, just make sure that only one of them can write!

And finaly you were [supposed to read the other discussions](https://github.com/search?q=repo%3Apschatzmann%2Farduino-audio-tools+pcm1808&type=discussions) about this topic
